### PR TITLE
kcollectd: init at 0.11.99.0

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change

new package
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/kcollectd/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+, kio
+, kiconthemes
+, kxmlgui
+, ki18n
+, kguiaddons
+, extra-cmake-modules
+, boost
+, shared_mime_info
+, rrdtool
+}:
```
Comment (doronbehar) [inline pkgs/tools/misc/kcollectd/default.nix:17]:
```suggestion
}:

```

File: pkgs/tools/misc/kcollectd/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+mkDerivation rec {
+  pname = "kcollectd";
+  version = "0.11.99.0";
```
Comment (doronbehar) [inline pkgs/tools/misc/kcollectd/default.nix:20]:
```suggestion
  version = "0.11.99.0";
  
```

File: pkgs/tools/misc/kcollectd/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  src = fetchFromGitLab {
+    owner = "aerusso";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0h4ymvzihzbmyv3z0bp28g94wxc6c7lgi3my0xbka3advxr811gn";
+  };
+
+  strictDeps = true;
+  nativeBuildInputs = [ cmake shared_mime_info ];
```
Comment (doronbehar) [inline pkgs/tools/misc/kcollectd/default.nix:29]: There's no such attribute `shared_mime_info`, that's why the ci error.

**Comment by doronbehar** [comment]
@symphorien Your PR should have eventually 1 commit:

```
<package-name>: init at <version>
```

**Comment by symphorien** [comment]
@GrahamcOfBorg build kcollectd

File: pkgs/tools/misc/kcollectd/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    kxmlgui
+    kiconthemes
+    ki18n
+    kguiaddons
+    boost
+    rrdtool
+  ];
+
+  meta = with lib; {
+    description = "A graphical frontend to collectd";
```
Comment (davidak) [inline pkgs/tools/misc/kcollectd/default.nix:47]:
```suggestion
    description = "A graphical frontend to collectd";
    homepage = "https://www.antonioerusso.com/projects/kcollectd/";
```

File: pkgs/tools/misc/kcollectd/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    maintainers = [ maintainers.symphorien ];
+    license = [ lib.licenses.gpl3Plus ];
```
Comment (davidak) [inline pkgs/tools/misc/kcollectd/default.nix:49]:
```suggestion
    license = [ lib.licenses.gpl3 ];
```

I only see https://gitlab.com/aerusso/kcollectd/-/blob/master/COPYING

**Review by davidak — CHANGES_REQUESTED** [review]

**Comment by davidak** [comment]
I built it locally and tried to start it, but get this error.
Screenshot from 2020-07-14 18-12-29
Is that expected? How is this package useful?

**Comment by doronbehar** [comment]
> Is that expected? How is this package useful?

Seems normal to me, if you are not running collectd.

**Comment by symphorien** [comment]
You have to specify `--basedir /path/to/collectd/rrd/files`

File: pkgs/tools/misc/kcollectd/default.nix (hunk 5 above)
Comment (symphorien) [reply pkgs/tools/misc/kcollectd/default.nix:49]: see the header of any source file, they specify "or any later version"

Comment (davidak) [reply pkgs/tools/misc/kcollectd/default.nix:50]: i see, nevermind

**Review by davidak — APPROVED** [review]
LGTM

File: pkgs/tools/misc/kcollectd/default.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
+{ lib
+, fetchFromGitLab
+, mkDerivation
+, fetchpatch
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/kcollectd/default.nix:4]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 93118` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kcollectd</li>
  </ul>
</details>

File: pkgs/tools/misc/kcollectd/default.nix (hunk 6 above)
Comment (symphorien) [reply pkgs/tools/misc/kcollectd/default.nix]: done thanks

**Review by doronbehar — APPROVED** [review]

**Pull request merged**